tests/annotations/tests.py

```diff
@@ -227,6 +313,11 @@ def test_distinct_on_with_annotation(self):
         ).distinct('name_len').values_list('name_len', flat=True)
         self.assertCountEqual(lengths, [3, 7, 8])
 
+    @skipUnlessDBFeature('can_distinct_on_fields')
+    def test_distinct_on_with_alias(self):
+        with self.assertRaisesMessage(FieldError, "Cannot resolve keyword 'other_rating' into field."):
```

Comment:
Should we also have a pointer here of the form

```suggestion
        with self.assertRaisesMessage(FieldError, "Cannot distinct on 'other_rating' alias. Use annotate to promote it"):
```

Reply:
I am not sure where this logic should go and what to do with other methods like `values`. 
The idea was to rely on existing code to error as if the field does not exist when trying to use the alias, but the aggregation failed with some cryptic message so I've added the correct error message.

Reply:
I improved messages for `values()` and `values_list()`. `distinct()` is more complicated because the exception is raised by `names_to_path()` that is used in many places, IMO we can leave the current message.